File: tests/fixtures/openai-cua/openai-cua-retry-transient-turn-errors-fixed/retry_loop_with_backoff.py
import time


def run_turn(client, input_items):
    for attempt in range(3):
        try:
            return client.responses.create(model="computer-use-preview", input=input_items)
        except Exception as exc:
            time.sleep(2**attempt)
    raise RuntimeError("turn failed after retries")

File: tests/fixtures/openai/openai-retry-transient-turn-errors-fixed/retry-loop-with-backoff.ts
export async function runTurn(client: any, model: string, input: any[]) {
  const maxAttempts = 3;
  for (let attempt = 0; attempt < maxAttempts; attempt += 1) {
    try {
      return await client.responses.create({ model, input, tools: [{ type: 'computer' }] });
    } catch (exc) {
      if (attempt === maxAttempts - 1) {
        return { success: false, error: String(exc) };
      }
      await new Promise((resolve) => setTimeout(resolve, 200 * (attempt + 1)));
    }
  }
}
